matron/src/device/device_list.c

```diff
@@ -149,6 +161,7 @@ static void dev_list_remove_node(struct dev_node *dn, union event_data *event_re
 }
 
 void dev_list_remove(device_t type, const char *node) {
+    pthread_mutex_lock(&dq.lock);
     struct dev_node *dn = dev_lookup_path(node, NULL);
     if (dn == NULL) {
         return;
```

Comment:
Mutex deadlock condition? Lock acquired on line-164, if potential return before free if `dn` is null. `dq` isn't referenced in this function directly , so the lock/unlock should probably go inside `dev_list_remove_node()`.

Reply:
More potential deadlocks in the switch statement for `DEV_TYPE_MIDI` and default.

Reply:
Good catch! Thanks